felix/dataplane/windows/endpoint_mgr.go

```diff
@@ -352,23 +351,75 @@ func (m *endpointManager) CompleteDeferredWork() error {
+						defaultTierEgressAppliesToEP = true
+					}
+					egressPolNames = append(egressPolNames, prependAll(policysets.PolicyNamePrefix, t.EgressPolicies))
+				}
+			}
+			log.Debugf("default tier has ingress policies: %v, egress policies: %v", defaultTierIngressAppliesToEP, defaultTierEgressAppliesToEP)
+
+			// If _no_ policies apply at all, then we fall through to the profiles.  Otherwise, there's no way to get
+			// from policies to profiles.
+			if len(ingressPolNames) == 0 || !defaultTierIngressAppliesToEP {
+				ingressPolNames = append(ingressPolNames, prependAll(policysets.ProfileNamePrefix, workload.ProfileIds))
+			}
+
+			if len(egressPolNames) == 0 || !defaultTierEgressAppliesToEP {
+				egressPolNames = append(egressPolNames, prependAll(policysets.ProfileNamePrefix, workload.ProfileIds))
 			}
 
-			if len(workload.Tiers) > 0 && len(workload.Tiers[0].EgressPolicies) > 0 {
-				logCxt.Debug("Workload Tier Policies will be applied Outbound")
-				outboundPolicyIds = append(outboundPolicyIds, prependAll(policysets.PolicyNamePrefix, workload.Tiers[0].EgressPolicies)...)
-			} else if len(workload.ProfileIds) > 0 {
-				logCxt.Debug("Profiles will be applied Outbound")
-				outboundPolicyIds = append(outboundPolicyIds, prependAll(policysets.ProfileNamePrefix, workload.ProfileIds)...)
+			// Expand the policies into rules.
+			var ingressRules, egressRules [][]*hns.ACLPolicy
+			for _, t := range ingressPolNames {
+				ingressRules = append(ingressRules, m.policysetsDataplane.GetPolicySetRules(t, true))
+			}
+			for _, t := range egressPolNames {
+				egressRules = append(egressRules, m.policysetsDataplane.GetPolicySetRules(t, false))
 			}
 
-			err := m.applyRules(id, endpointId, inboundPolicyIds, outboundPolicyIds)
+			// Flatten any tiers.
+			flatIngressRules := flattenTiers(ingressRules)
+			flatEgressRules := flattenTiers(egressRules)
+
+			if log.GetLevel() >= log.DebugLevel {
+				for _, rule := range flatIngressRules {
+					log.WithFields(log.Fields{"rule": rule}).Debug("ingress rules after flattening")
+				}
+				for _, rule := range flatEgressRules {
+					log.WithFields(log.Fields{"rule": rule}).Debug("egress rules after flattening")
+				}
+			}
+
+			// Make sure priorities are ascending.
+			rewritePriorities(flatIngressRules, policysets.PolicyRuleMaxPriority)
+			rewritePriorities(flatEgressRules, policysets.PolicyRuleMaxPriority)
+
+			// Finally, add default allow rule with a host-scope to allow traffic through
+			// the host windows firewall. Required by l2bridge network.
+			// We need to add host rules after priority is rewritten because the value of the priority
+			// for a host rule depends on AclNoHostRulePriority feature support.
+			flatIngressRules = append(flatIngressRules, m.policysetsDataplane.NewHostRule(true))
+			flatEgressRules = append(flatEgressRules, m.policysetsDataplane.NewHostRule(false))
+
+			err := m.applyRules(id, endpointId, flatIngressRules, flatEgressRules)
```

Comment:
I now see that this is what replaces the host rule removed from policysets.go :)

Reply:
yeah, exactly.